### PR TITLE
klippy: 2022-03-14 -> 2022-06-18; drop python2

### PR DESCRIPTION
**Pull request by cab404**
###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by cab404** [comment]
Yes, it does work on python3, alas maintainer didn't yet update shebangs.

File: pkgs/servers/klipper/default.nix [hunk 1]
```diff
@@ -1,32 +1,35 @@
...
+    rev = "d3c4ba4839dd7a4339ae024752e6c6424884c185";
+    sha256 = "sha256-2Fq56JIk5qcKpWffz1k/EJ+xYAnUpuxvCryq88l//8E=";
   };
 
   sourceRoot = "source/klippy";
 
-  # there is currently an attempt at moving it to Python 3, but it will remain
-  # Python 2 for the foreseeable future.
-  # c.f. https://github.com/KevinOConnor/klipper/pull/3278
-  # NB: This is needed for the postBuild step
```
Comment (lovesegfault) [inline pkgs/servers/klipper/default.nix:23]: Please keep the nota bene.

Comment (cab404) [reply pkgs/servers/klipper/default.nix:23]: Done, added a note below. Why do you need this?

Comment (cab404) [reply pkgs/servers/klipper/default.nix:23]: sure, one sec

File: pkgs/servers/klipper/default.nix [hunk 2]
```diff
@@ -1,32 +1,38 @@
...
-  buildInputs = [ (python2.withPackages (p: with p; [ cffi pyserial greenlet jinja2 numpy ])) ];
+  buildInputs = [ (python3.withPackages (p: with p; [ cffi pyserial greenlet jinja2 markupsafe ])) ];
 
   # we need to run this to prebuild the chelper.
-  postBuild = "python2 ./chelper/__init__.py";
+  postBuild = "python ./chelper/__init__.py";
+
+  # 2022-06-28: Python 3 is already supported in klipper, alas shebangs remained
+  # the same - we replace them in patchPhase.
+  patchPhase = ''
```
Comment (SuperSandro2000) [inline pkgs/servers/klipper/default.nix:30]: patchPhase must not be ovewritten as it breaks patches. Please use postPatch instead.

Comment (cab404) [reply pkgs/servers/klipper/default.nix:30]: Sorry!

Will fix in the next klipper patchset

**Pull request merged**